### PR TITLE
fix(core): checkout template repository by revision

### LINKED ISSUE #2024: Initializing from a template repo commit sha fails
This should clone the repo and check out the correct commit instead of trying to check out the ref:

```
renku init -s https://github.com/SwissDataScienceCenter/renku-project-template -r aef231b test-template-update2
Fetching template from https://github.com/SwissDataScienceCenter/renku-project-template@aef231b...
Error: Cannot fetch and checkout reference aef231b
Traceback (most recent call last):
  File "/Users/rok/Projects/renku-core/renku-python/renku/core/commands/init.py", line 266, in fetch_template_from_git
    template_repo.remotes.origin.fetch(ref)
  File "/Users/rok/.local/pipx/venvs/renku/lib/python3.9/site-packages/git/remote.py", line 797, in fetch
    res = self._get_fetch_info_from_stderr(proc, progress)
  File "/Users/rok/.local/pipx/venvs/renku/lib/python3.9/site-packages/git/remote.py", line 676, in _get_fetch_info_from_stderr
    proc.wait(stderr=stderr_text)
  File "/Users/rok/.local/pipx/venvs/renku/lib/python3.9/site-packages/git/cmd.py", line 408, in wait
    raise GitCommandError(self.args, status, errstr)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git fetch -v origin aef231b
  stderr: 'fatal: couldn't find remote ref aef231b'
```

possibly related to #1988

### PR DESCRIPTION
**Pull request by vigsterkr**
# Description

Fixes #2024

### CODE REVIEW

File: renku/core/commands/init.py [hunk 1]
```diff
@@ -353,7 +353,7 @@ def fetch_template_from_git(source, ref="master", tempdir=None):
 
     try:
         # fetch ref and set the HEAD
-        template_repo.remotes.origin.fetch(ref)
+        template_repo.remotes.origin.fetch()
```
Comment (vigsterkr) [inline renku/core/commands/init.py:356]: @rokroskar @Panaetius note this only adds support for commit sha. support for tag would still fail. personally having a support for checking out a tag would be nice. but i'm wondering whether you think that'd would be as well desirable.

Comment (vigsterkr) [reply renku/core/commands/init.py:356]: actually based on the docstring this function advertises that
```
    :param ref: reference for git checkout - branch, commit or tag
```
so i'll fix it for tag as well.

**Review by Panaetius — COMMENTED** [review]
Should we add a test for this?

**Comment by vigsterkr** [comment]
> Should we add a test for this?

like the 1st commit sha of the template repo?

**Review by Panaetius — APPROVED** [review]
Thanks! :+1:

**Pull request merged**